pumpkin-lib/src/engine/cp/propagation/mod.rs

```diff
@@ -24,20 +24,16 @@
 //!
 //! # Practical
 //!
-//! Each concrete propagator is associated with two traits:
-//! - [`Propagator`]: contains the propagator logic.
-//! - [`PropagatorConstructor`]: provides the functionality to register variables for their
-//!   corresponding [`DomainEvents`] with the solver. Note that structs implementing this trait do
-//!   not directly communicate with the solver, but rather use the [`PropagatorConstructorContext`]
-//!   as a communication point during creation to get access to variables. *Note that this trait is
-//!   **not** implemented on the propagator struct itself but rather on another struct holding the
-//!   arguments necessary for creating the propagator; for an example, see
-//!   [`LinearLessOrEqualConstructor`] and [`LinearLessOrEqualPropagator`]!*
-//!
-//! When domain changes happen for a variable outside the propagator, the propagator will receive
-//! information that its variable with a specific [`LocalId`] has changed (see
-//! [`Propagator::notify`]). The idea behind using the structs apart from Propagator is to support
-//! views \[2\] on variables.
+//! Each concrete propagator is associated with one trait; [`Propagator`]: contains the propagator
```

Comment:
This paragraph can be more thoroughly rewritten. Of course an implementation of propagator contains the propagator logic.